spec/smtp_protocol_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require 'async/io/protocol/line'
+require 'async/io/socket'
+require 'mail_catcher'
+require 'net/smtp'
+
+message = <<~END_OF_MESSAGE
+    From: Your Name <your@mail.address>
+    To: Destination Address <someone@example.com>
+    Subject: test message
+    Date: Sat, 23 Jun 2001 16:26:43 +0900
+    Message-Id: <unique.message.id.string@example.com>
+  #{'  '}
+    This is a test message.
+END_OF_MESSAGE
```

Comment:
This doesn't yield a valid message, the indentation becomes odd:

```ruby
message = <<~END_OF_MESSAGE
     From: Your Name <your@mail.address>
     To: Destination Address <someone@example.com>
     Subject: test message
     Date: Sat, 23 Jun 2001 16:26:43 +0900
     Message-Id: <unique.message.id.string@example.com>
   #{'  '}
     This is a test message.
 END_OF_MESSAGE

puts message.pretty_inspect
```
```ruby
"  From: Your Name <your@mail.address>\n" +
"  To: Destination Address <someone@example.com>\n" +
"  Subject: test message\n" +
"  Date: Sat, 23 Jun 2001 16:26:43 +0900\n" +
"  Message-Id: <unique.message.id.string@example.com>\n" +
"  \n" +
"  This is a test message.\n"
```

Was this intentional? Should it be:

```suggestion
message = <<~MESSAGE
  From: Your Name <your@mail.address>
  To: Destination Address <someone@example.com>
  Subject: test message
  Date: Sat, 23 Jun 2001 16:26:43 +0900
  Message-Id: <unique.message.id.string@example.com>

  This is a test message.
MESSAGE
```
```ruby
puts message.pretty_inspect
```
```ruby
"From: Your Name <your@mail.address>\n" +
"To: Destination Address <someone@example.com>\n" +
"Subject: test message\n" +
"Date: Sat, 23 Jun 2001 16:26:43 +0900\n" +
"Message-Id: <unique.message.id.string@example.com>\n" +
"\n" +
"This is a test message.\n"
```

Reply:
I think space was added by the `rubocop -A` command. I'll fix it.